client/src/analytics/index.ts

```diff
@@ -0,0 +1,18 @@
+import TagManager from 'react-gtm-module';
+
+import {
+  devAnalyticsId,
+  prodAnalyticsId
+} from '../../../config/analytics-settings';
+
+import envData from '../../../config/env.json';
+
+const { deploymentEnv } = envData;
+
+const gtmId = deploymentEnv === 'staging' ? devAnalyticsId : prodAnalyticsId;
+
+if (typeof document !== `undefined`) {
+  TagManager.initialize({ gtmId });
+}
```

Comment:
```suggestion
TagManager.initialize({ gtmId });
```

Just wondering, is it possible for the document API to be `undefined`?

Reply:
Yes — it is possible. When not in a browser context. This check should be there.